UISchema API docs (#3603)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/uischema/index.md

#### Error Responses

If an invalid UI Schema ID is included in the request, the response returns a 404 NOT FOUND with error code E0000008.

## UI Schema API objects

### UI Schema Request object

#### UI Schema Request properties

| Property   | Type | Description |
| ---------- | ------------------------------------- | ----------------------------------------------------------- |
| `id`       | String                                | The unique ID of the ui schema |
| `uiSchema` | [UI Schema Object](#ui-schema-object) | N/A |

#### UI Schema Request example

### UI Schema Response object

#### UI Schema Response object properties

| Property      | Type                                                                    | Description                                                                                                       |
| ------------- | ----------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------- |
| `id`          | String                                                                  | The unique ID of the UI schema |

> | `uiSchema`    | See [UI Schema Object](#ui-schema-object). | N/A |

Reviewer: ```suggestion
| `uiSchema`    | See [UI Schema Object](#ui-schema-object) | N/A |
```

| `lastUpdated` | String (ISO-8601) | Timestamp when the UI Schema was last modified |

> | `_links`      | Object ([JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)) | Discoverable resources related to the UISchema |

Reviewer: is it UI Schema or is it `uiSchema`?
```suggestion
| `_links`      | Object ([JSON HAL](http://tools.ietf.org/html/draft-kelly-json-hal-06)) | Discoverable resources related to the UI Schema |
```

#### UI Schema Response object example

```json
{
  "id": "uis4a97f4pmZsdxKu0g7",
  "uiSchema": {},
  "created": "2022-07-28T13:02:25.000Z",
  "lastUpdated": "2022-07-28T14:31:42.000Z",
  "_links": {
    "self": {
      "href": "https://dev-jake-2.trexcloud.com/api/v1/meta/uischemas/uis4a97f4pmZsdxKu0g7",
      "hints": {
        "allow": ["GET", "PUT", "DELETE"]
      }
    }
  }
}
```

### UI Schema object

#### UI Schema properties

| Property      | Type   | Description |
| ------------- | ------ | ----------- |

> | `buttonLabel` | String | Specifies the button label for the `submit` button at the bottom of the enrollment form. Defaults to `Submit`. |

Reviewer: ```suggestion
| `buttonLabel` | String | Specifies the button label for the `Submit` button at the bottom of the enrollment form. Defaults to `Submit`. |
```

> | `elements`    | Array  |  |

Author: @jakebacher2-okta There are a few properties in the "UI Schema object" section that are missing descriptions. They still had the template placeholder content which I deleted.

Reviewer 2: Not sure what the ask is here. If it is for `elements` missing it's description, we should point it to the `### UI Schema Element object` section below I believe.

Author: Ah. Thank you!

> | `elements`    | Array  | See [UI Schema Element object](#ui-schema-element-object). |

Reviewer: ```suggestion
| `elements`    | Array  | See [UI Schema Element object](#ui-schema-element-object) |
```

| `label`       | String | Specifies the label at the top of the enrollment form under the logo. Defaults to `Sign in`.  |
| `type`        | String | Type of layout  |

#### UI Schema example

```json
{
  "buttonLabel": "Submit",
  "elements": [],
  "label": "Sign in",

>   "type": "Group

Reviewer: ```suggestion
  "type": "Group"
```

}
```

### UI Schema Element object

The UI Schema Element object specifies the configuration of an input field on an enrollment form.

#### UI Schema Element properties

| Property | Type | Description | Supported Values |
| -------- | ---- | ------ | ------- |

> | `buttonLabel` | String  | Specifies the button label for the `submit` button at the bottom of the enrollment form. Defaults to `Submit`. | N/A |

Reviewer: ```suggestion
| `buttonLabel` | String  | Specifies the button label for the `Submit` button at the bottom of the enrollment form. Defaults to `Submit`. | N/A |
```

> | `options`     | [UI Schema Element Options ](#ui-schema-element-options-object) |  |

Author: Needs description.

> | `scope`       | String | Specifies the property to which the input field should be bound. Must follow the format `#/properties/PROPERTY_NAME` where `PROPERTY_NAME` is a variable name for an attribute in `profile editor`. | N/A              |

Reviewer: ```suggestion
| `scope`       | String | Specifies the property to which the input field should be bound. It must follow the format `#/properties/PROPERTY_NAME` where `PROPERTY_NAME` is a variable name for an attribute in `profile editor`. | N/A              |
```

| `type`        | String  | Specifies the relationship between this input element and `Scope`. The `Control` value specifies that this input controls the value represented by `scope`. | `Control` |

#### UI Schema Element example

```json
{
  "label": "Primary email",
  "options": {},
  "scope": "#/properties/email",
  "type": "Control"
}
```

### UI Schema Element Options object

> The UI Schema Element Options object specifies how the input is displayed. The supported formats are:

Reviewer: ```suggestion
The UI Schema Element Options object specifies how the input appears. The supported formats are:
```

- `text`: Text input. This is the default format for the majority of property types.
- `radio`: Radio button options. This option is only available for `string` data types with an `enum` or `one of` constraint.

> - `select`: Displays input as a dropdown. This option is only available for the `country-code` data type or a `string` data type with an `enum` or `one of` constraint.

Reviewer: ```suggestion
- `select`: Displays input as a dropdown list. This option is only available for the `country-code` data type or a `string` data type with an `enum` or `one of` constraint.
```

- `checkbox`: Displays input as a checkbox. This option is only available for `boolean` data types.

> - `radio_yes_no`: Displays input as 2 radio buttons, one with the option `yes` and the other `no`. This option is only available for `boolean` data types.

Reviewer: ```suggestion
- `radio_yes_no`: Displays input as two radio buttons, one with the option `yes` and the other `no`. This option is only available for `boolean` data types.
```

> - `radio_true_false`: Displays input as 2 radio buttons, one with the option `true` and the other `false`. This option is only available for `boolean` data types.

Reviewer: ```suggestion
- `radio_true_false`: Displays input as two radio buttons, one with the option `true` and the other `false`. This option is only available for `boolean` data types.
```

#### UI Schema Element Option properties

| Property | Type   | Description                                 | Supported Values                                                          |
| -------- | ------ | ------------------------------------------- | ------------------------------------------------------------------------- |

> | `format` | String | Specifies how the input should be displayed | `text`, `radio`, `select`, `checkbox`, `radio_yes_no`, `radio_true_false` |

Reviewer: ```suggestion
| `format` | String | Specifies how the input should appear | `text`, `radio`, `select`, `checkbox`, `radio_yes_no`, `radio_true_false` |
```

---
title: UI Schema API
category: management
---

# UI Schema API

<ApiLifecycle access="ie" />

> The Okta UI Schema API allows you to control how inputs appear on a form.

Author: I think that we need to clarify which forms they can update.
